tests/test_hierarchy.cc

```diff
@@ -193,35 +195,51 @@ BOOST_DATA_TEST_CASE(
+    ref_solution[std::make_tuple("hyper_cube" , true  , "Reverse Cuthill_McKee" , "lapack" , true)]   = 0.1575262224;
+    ref_solution[std::make_tuple("hyper_ball" , false , "None"                  , "lapack" , false)]  = 0.1146629782;
+    ref_solution[std::make_tuple("hyper_ball" , false , "None"                  , "lapack" , true)]   = 0.3022004744;
+    ref_solution[std::make_tuple("hyper_ball" , false , "Reverse Cuthill_McKee" , "lapack" , false)]  = 0.1146629782;
+    ref_solution[std::make_tuple("hyper_ball" , false , "Reverse Cuthill_McKee" , "lapack" , true)]   = 0.3022004744;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "None"                  , "lapack" , false)]  = 0.1024334788;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "None"                  , "lapack" , true)]   = 0.2977841482;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "Reverse Cuthill_McKee" , "lapack" , false)]  = 0.1024334788;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "Reverse Cuthill_McKee" , "lapack" , true)]   = 0.2977841482;
+
+    ref_solution[std::make_tuple("hyper_cube" , false , "None"                  , "lanczos" , false)] = 0.0491724046;
+    ref_solution[std::make_tuple("hyper_cube" , false , "None"                  , "lanczos" , true)]  = 0.1482630509;
+    ref_solution[std::make_tuple("hyper_cube" , false , "Reverse Cuthill_McKee" , "lanczos" , false)] = 0.0491724046;
+    ref_solution[std::make_tuple("hyper_cube" , false , "Reverse Cuthill_McKee" , "lanczos" , true)]  = 0.1482630509;
+    ref_solution[std::make_tuple("hyper_cube" , true  , "None"                  , "lanczos" , false)] = 0.0488968016;
+    ref_solution[std::make_tuple("hyper_cube" , true  , "None"                  , "lanczos" , true)]  = 0.1575200977;
+    ref_solution[std::make_tuple("hyper_cube" , true  , "Reverse Cuthill_McKee" , "lanczos" , false)] = 0.0488968016;
+    ref_solution[std::make_tuple("hyper_cube" , true  , "Reverse Cuthill_McKee" , "lanczos" , true)]  = 0.1575200977;
+    ref_solution[std::make_tuple("hyper_ball" , false , "None"                  , "lanczos" , false)] = 0.0953745739;
+    ref_solution[std::make_tuple("hyper_ball" , false , "None"                  , "lanczos" , true)]  = 0.3017812295;
+    ref_solution[std::make_tuple("hyper_ball" , false , "Reverse Cuthill_McKee" , "lanczos" , false)] = 0.0953745739;
+    ref_solution[std::make_tuple("hyper_ball" , false , "Reverse Cuthill_McKee" , "lanczos" , true)]  = 0.3017812295;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "None"                  , "lanczos" , false)] = 0.0966058675;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "None"                  , "lanczos" , true)]  = 0.2982453175;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "Reverse Cuthill_McKee" , "lanczos" , false)] = 0.0966058675;
+    ref_solution[std::make_tuple("hyper_ball" , true  , "Reverse Cuthill_McKee" , "lanczos" , true)]  = 0.2982453175;
```

Comment:
TL;DR
Would you confirm that you did add convergences rates for Lanczos and that nothing else change?  Can you comment on the new gold values?  Are they reasonably close to their LAPACK's counterpart?

Reply:
Yes. The values are really close but not sure how robust the thing is. Lanczos test now fails, so I may need to play around with things. The whole thing is annoying, but proper fix is unclear.